drivers/sensor/rpi_pico_temp/rpi_pico_temp.c

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright (c) 2023 TOKITA Hiroshi
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/device.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/drivers/adc.h>
+#include <zephyr/logging/log.h>
+
+#include <hardware/adc.h>
+
+LOG_MODULE_REGISTER(rpi_pico_temp, CONFIG_SENSOR_LOG_LEVEL);
+
+#define DT_DRV_COMPAT raspberrypi_pico_temp
+
+struct rpi_pico_temp_config {
+	const struct device *adc;
+	const struct adc_channel_cfg ch_cfg;
+};
+
+struct rpi_pico_temp_data {
+	struct adc_sequence adc_seq;
+	int16_t sample;
+};
+
+static int rpi_pico_temp_sample_fetch(const struct device *dev, enum sensor_channel chan)
+{
+	const struct rpi_pico_temp_config *cfg = dev->config;
+	struct rpi_pico_temp_data *data = dev->data;
+	int rc;
+
+	if (chan != SENSOR_CHAN_ALL && chan != SENSOR_CHAN_DIE_TEMP) {
+		return -ENOTSUP;
+	}
+
+	rc = adc_channel_setup(cfg->adc, &cfg->ch_cfg);
+	if (rc) {
+		LOG_DBG("Setup AIN%u got %d", cfg->ch_cfg.channel_id, rc);
```

Comment:
AIN should be replaced by a more pico suitable term

Reply:
Change it to `ADC channel`